entire_vds_pipeline.py

```diff
@@ -414,8 +416,28 @@ def export_to_elasticsearch(
 logger.info("\n==> create HailContext")
 hc = hail.HailContext(log="/hail.log")
 
+logger.info("Reading in dataset...")
 vds = read_in_dataset(input_path, args.datatype, filter_interval)
 
+#NOTE: if sample IDs are remapped first thing, then the fam file should contain the desired (not original IDs)
+if args.remap_sample_ids:
+    logger.info("Remapping sample ids...")
+    id_map = hc.import_table(args.remap_sample_ids, impute=True, no_header=True)
+    mapping = dict(zip(id_map.query('f0.collect()'), id_map.query('f1.collect()')))
+    vds = vds.rename_samples(mapping)
+    logger.info('Remapped {} sample ids...'.format(id_map.count()))
+
+
+# subset samples as desired
+if args.subset_samples:
+    logger.info("Subsetting to specified samples...")
+    keep_samples = hc.import_table(args.subset_samples, impute=True, no_header=True).key_by('f0')
+    original_sample_count, original_var_count = vds.count()
```

Comment:
vds.count() can take a long time for large callsets. Maybe better to use vds.sample_ids or vds.num_samples.